### PR TITLE
feat(14063): Create map Breadcrumbs component

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/My-tasks-3575#Task/Create-map-breadcrumbs-component-14063

image

## Release Notes

- **New Features**
  - Introduced a `BreadcrumbsPanel` for enhanced navigation within the map interface.
  - Added `ADMIN_BOUNDARY_BREADCRUMBS` feature for improved administrative boundary management.
  - Launched the `Ellipsis` component to manage overflow breadcrumb items, improving access to hidden navigation options.

- **Improvements**
  - Updated the `@konturio/ui-kit` package for better performance and stability.
  - Enhanced breadcrumb navigation with responsive styling and improved visibility for hidden items.

- **Styles**
  - Added new CSS classes for breadcrumb items and layout adjustments for better responsiveness across devices.

### CODE REVIEW

**Comment by alesiahil** [comment]
I don't see a crosshair in the center of the map but it's not critical.

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  const [items] = useAtom(breadcrumbsItemsAtom);
+
+  const handleClick = useAction(onBreadcrumbClick);
+
+  return items.length > 0 ? (
+    <Panel resize="none" className={s.breadcrumbsPanel}>
+      <Breadcrumbs items={items} onClick={handleClick} />
+    </Panel>
+  ) : (
+    <></>
```
Comment (Natallia-Harshunova) [inline src/features/breadcrumbs/BreadcrumbsPanel.tsx]: As far as I remember, it better to be `null` when empty render

File: src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { action, atom } from '@reatom/core';
+import type { BoundaryOption } from '~utils/map/boundaries';
+
+export const breadcrumbsItemsAtom = atom<BoundaryOption[]>([], 'breadcrumbsAtom');
+
+export const onBreadcrumbClick = action((ctx, value: string | number) => {
+  const items = ctx.get(breadcrumbsItemsAtom);
+  const index = items.findIndex((item) => item.value === value);
+  if (index !== -1 && index < items.length) {
```
Comment (Natallia-Harshunova) [inline src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts]: Does the second check is needed here? i think `tems.findIndex` cannot return index greater then length of items

File: src/features/breadcrumbs/components/commonStyles.module.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.separator {
+  margin: 0 4px;
```
Comment (Natallia-Harshunova) [inline src/features/breadcrumbs/components/commonStyles.module.css]: `0 var(--half-unit)`

File: src/views/Map/Layouts/Layout.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { Text } from '@konturio/ui-kit';
```
Comment (Natallia-Harshunova) [inline src/views/Map/Layouts/Layout.tsx]: Unused import

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/features/breadcrumbs/BreadcrumbsPanel.tsx (hunk 1 above)
Comment (albaranau) [reply src/features/breadcrumbs/BreadcrumbsPanel.tsx]: Fixed

File: src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts (hunk 2 above)
Comment (albaranau) [reply src/features/breadcrumbs/atoms/breadcrumbsItemsAtom.ts]: Fixed

File: src/views/Map/Layouts/Layout.tsx (hunk 4 above)
Comment (albaranau) [reply src/views/Map/Layouts/Layout.tsx]: Fixed

File: src/features/breadcrumbs/components/commonStyles.module.css (hunk 3 above)
Comment (albaranau) [reply src/features/breadcrumbs/components/commonStyles.module.css]: Fixed

File: src/views/Map/Map.tsx [hunk 5]
```diff
@@ -136,6 +137,9 @@ export function MapPage() {
           layersAndLegends={<LayersAndLegends />}
           matrix={<></>}
           timeline={featureFlags[FeatureFlag.EPISODES_TIMELINE] && <EventEpisodes />}
+          breadcrumbs={
+            featureFlags[FeatureFlag.ADMIN_BOUNDARY_BREADCRUMBS] && <BreadcrumbsPanel />
+          }
```
[the remark this answers is not in the document]
Comment (albaranau) [reply src/views/Map/Map.tsx:142]: What exactly made this prop inconsistent with the other components' ones?

**Pull request merged**